README.md

```diff
@@ -1,28 +1,142 @@
 # IOS Vast Client
 
-IOS Vast Client is a Swift Framework which implements the [VAST 3.0](https://www.iab.com/guidelines/digital-video-ad-serving-template-vast-3-0/) spec.
+IOS Vast Client is a Swift Framework which implements the [VAST 4.0](https://www.iab.com/guidelines/digital-video-ad-serving-template-vast/) spec.
 
 This project is ongoing and in very early stages of development. As this project is being developed we will only target a subset of the VAST spec to meet our immediate needs. We do not recommend to use this library at this time. 
 
 ## Features
 
-* VAST 3.0 Spec Complient
+* VAST 4.0 Spec Complient
 * Vast XML Parser and Validator
 * VAST Impression and Ad Tracking
 * VAST Error Tracking
 
 ## API docs
 
-TODO
+iOS Vast Client contains 2 interfaces for public use.
+
+### VastClient
+
+Fetch and parse VAST file from URL. You can customize behaviour with `VastClientOptions` during initialization of `VastClient`.
+
+`VastClient` result is `VastModel` structure or `Error` in case the fetch failed. You should pass this model to `VastTracker`.
+
+### VastTracker
+
+Handels tracking of ad break progress and errors. Notifies state updates via delegate calls.
+
+#### Initialization
+
+Init `VastTracker` with `VastModel` structure but always make sure to use the actual `VastAd` and other information provided via delegate function calls - do not keep the `VastModel` as not all information from the `VastModel` might be valid for playback etc.
+
+If you initialize `VastTracker` with delegate - `func adBreakStart(vastTracker: VastTracker, totalAds: Int)` will be called immediately. 
+
+#### Progress
+
+`func updateProgress(time: Double) throws`
+After you initialize `VastTracker` you should call this function with `time = 0` parameter to start tracking ad playback. `VastTracker` will select linear ad to play and will notify you via delegate call `func adStart(vastTracker: VastTracker, ad: VastAd)`
+
+You have to call this function periodically during ad playback.
+These delegate function will be called during normal ad playback and you do not have to react to them - they are only informative
+
+```swift
+func adFirstQuartile(vastTracker: VastTracker, ad: VastAd)
+func adMidpoint(vastTracker: VastTracker, ad: VastAd)
+func adThirdQuartile(vastTracker: VastTracker, ad: VastAd)
+```
+
+#### Ad End
+
+When ad playback ends your app is responsible for calling `finishedPlayback()`.
```

Comment:
So the intended purpose is to force the user to call finishedPlayback after each ad in a pod? Whats the driver behind this decision?

Reply:
Reason behind this was that ads were sometimes cut before the real playback finished.
Duration received in Vast file is integer (or usually is) but the asset might not finish exactly with .0

When you have ad that had Vast duration of 14 but asset play time was 14.9 it was cut 0.9 seconds, because the tracker reported that ad has finished (and new ad has started to play)

If you had duration of 14 and ads was really only 13.9, the final tracking never happened and no further ads were played.

I am very open to discussing this to get better functioning client!

Reply:
This is an interesting issue. I need to think on this for a little bit.